jetbrains-core/src/software/aws/toolkits/jetbrains/core/AwsSettingsPanel.kt

```diff
@@ -220,10 +222,21 @@ private class ChangeCredentialsAction(val credentialsProvider: ToolkitCredential
 
     override fun setSelected(e: AnActionEvent, selected: Boolean) {
         if (selected) {
-            getAccountSetting(e).activeCredentialProvider = credentialsProvider
+            if (!credentialsProvider.isValid(AwsSdkClient.getInstance().sdkHttpClient)) {
+                notifyWarn(
+                    title = message("credentials.invalid.title"),
+                    content = message("credentials.invalid.notification", credentialsProvider.displayName),
+                    listener = NotificationListener { notification, _ ->
+                        ActionManager.getInstance().getAction("aws.settings.upsertCredentials").actionPerformed(e)
```

Comment:
I think it would be better to just add this Action to the notification using `Notification.addAction`

Reply:
We can use `NotificationAction` as a wrapper of the `CreateOrUpdateCredentialProfilesAction`